portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +40,18 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
     // adds comments to datastore 
     String comment = request.getParameter("comments");
+    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     if(comment != null && !comment.equals("")) {
-      Entity taskEntity = new Entity("Task");
+      Entity taskEntity = new Entity("userInfo");
```

Comment:
I think this should probably be named Comment.

We want the datastore entity name to represent the concept (and user vs. comment is different. I think entity names are case sensitive (so this is different than the UserInfo you use in AuthenticationServlet); even if they aren't, Datastore queries will only return entities that match only those properties. so this is effectively returning just comments (no nicknames)

See:
https://cloud.google.com/datastore/docs/concepts/queries#restrictions_on_queries